testutil/blackhole_watcher.go

```diff
@@ -0,0 +1,48 @@
+package testutil
+
+import (
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"golang.org/x/net/context"
+)
+
+// MockWatcher mocks an etcd client that just blackholes a few watch requests
+type MockWatcher struct {
```

Comment:
If the package is called `mocks`, this could just be `Watcher` and the filename is `watcher.go`